src/plugins/notebook/components/NotebookEmbed.vue

```diff
@@ -88,38 +87,88 @@ export default {
-                callback: this.getRemoveDialog.bind(this)
+        showMenuItems(event) {
+            const x = event.x;
+            const y = event.y;
+
+            const menuOptions = {
+                menuClass: 'c-ne__embed__actions-menu',
+                placement: this.openmct.menus.menuPlacement.TOP_RIGHT
+            };
+
+            this.openmct.menus.showSuperMenu(x, y, this.menuActions, menuOptions);
+        },
+        addMenuActions() {
+            if (this.embed.snapshot) {
+                const viewSnapshot = {
+                    id: 'viewSnapshot',
+                    cssClass: 'icon-camera',
+                    name: 'View Snapshot',
+                    description: 'View the snapshot image taken in the form of a jpeg.',
+                    onItemClicked: () => this.openSnapshot()
+                };
+
+                this.menuActions = [viewSnapshot];
+            }
+
+            const navigateToItem = {
+                id: 'navigateToItem',
+                cssClass: this.embed.cssClass,
+                name: 'Navigate to Item',
+                description: 'Navigate to the item with the current time settings.',
+                onItemClicked: () => this.navigateToItem()
             };
-            const preview = {
-                id: 'preview',
+
+            const navigateToItemInTime = {
+                id: 'navigateToItemInTime',
+                cssClass: this.embed.cssClass,
+                name: 'Navigate to Item in Time',
+                description: 'Navigate to the item in its time frame when captured.',
+                onItemClicked: () => this.navigateToItemInTime()
+            };
+
+            const quickView = {
+                id: 'quickView',
                 cssClass: 'icon-eye-open',
-                name: 'Preview',
-                callback: this.previewEmbed.bind(this)
+                name: 'Quick View',
+                description: 'Full screen overlay view of the item.',
+                onItemClicked: () => this.previewEmbed()
             };
 
-            this.popupMenuItems = [preview];
+            this.menuActions = this.menuActions.concat([quickView, navigateToItem, navigateToItemInTime]);
 
             if (!this.isLocked) {
-                this.popupMenuItems.unshift(removeEmbed);
+                const removeEmbed = {
+                    id: 'removeEmbed',
+                    cssClass: 'icon-trash',
+                    name: this.removeActionString,
+                    description: 'Permanently delete this embed from this Notebook entry.',
+                    onItemClicked: this.getRemoveDialog.bind(this)
+                };
+
+                this.menuActions.push(removeEmbed);
             }
 
         },
+        async setEmbedObjectPath() {
+            this.objectPath = await this.openmct.objects.getOriginalPath(this.embed.domainObject.identifier);
+
+            if (this.objectPath[this.objectPath.length - 1].type === 'root') {
```

Comment:
Is the length of `this.objectPath` guaranteed to be at least `1`? If not, this will break.

Reply:
not entirely sure to be honest, but to be safe, I put a check in there.